File: suites/03_data_structures/comparisons/linked_list.go
package main
import "fmt"
func pointerChase(nodes, rounds int64) int64 {
	cursor, checksum := int64(1), int64(0)
	for s := int64(0); s < rounds; s++ {
		cursor = ((cursor * 1103515245) + 12345) % nodes
		checksum += cursor
	}
	return checksum
}
func main() { fmt.Println(pointerChase(1000003, 8000000)) }

File: suites/03_data_structures/comparisons/linked_list.py
def pointer_chase(nodes: int, rounds: int) -> int:
    cursor, checksum = 1, 0
    for _ in range(rounds):
        cursor = ((cursor * 1103515245) + 12345) % nodes
        checksum += cursor
    return checksum
if __name__ == "__main__":
    print(pointer_chase(1000003, 8000000))
